### PR TITLE
Move Replace Disk to middlewared

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/disk.py [hunk 1]
```diff
@@ -248,6 +246,43 @@ def geli_attach_single(self, dev, key, passphrase=None, skip_existing=False):
...
+    @private
+    def geli_testkey(self, pool, passphrase):
+        """
+        Test key for geli providers of a given pool
+
+        Returns:
+            bool
+        """
+
+        with tempfile.NamedTemporaryFile(mode='w+', dir='/tmp', delete=False) as tf:
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/disk.py]: Why `delete=False`?

File: src/middlewared/middlewared/plugins/pool.py [hunk 2]
```diff
@@ -692,6 +695,132 @@ def disk_to_stripe():
...
+
+        unused_disks = await self.middleware.call('disk.get_unused')
+        disk = list(filter(lambda x: x['identifier'] == options['disk'], unused_disks))
+        if not disk:
+            verrors.add('options.disk', 'Disk not found.', errno.ENOENT)
+        else:
+            disk = disk[0]
+
+        if not options['force'] and not await self.middleware.call(
+            'disk.check_clean', disk['devname']
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py]: `disk` might be `None` here, probably we should have raised `verrors` above

File: src/middlewared/middlewared/plugins/pool.py [hunk 3]
```diff
@@ -692,6 +695,132 @@ def disk_to_stripe():
...
+            # is initiated.
+            if options['label'].isdigit():
+                await self.middleware.call('zfs.pool.detach', pool['name'], options['label'])
+        except Exception as e:
+            try:
+                # If replace has failed lets detach geli to not keep disk busy
+                await self.middleware.call('disk.geli_detach', new_devname)
+            except Exception:
+                self.logger.warn(f'Failed to geli detach {new_devname}', exc_info=True)
+            raise e
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py]: Did you mean just `raise`?

File: src/middlewared/middlewared/plugins/pool.py (hunk 2 above)
Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py]: True, fixing!

**Comment by william-gr** [comment]
@themylogin two comments were handled, let me know if you agree with the third. Thanks!

**Pull request merged**